pkg/controller/persistentvolume/types.go

```diff
@@ -25,6 +25,13 @@ import (
 	"k8s.io/kubernetes/pkg/client/cache"
 )
 
+const (
+	// A PV created specifically for one claim must contain this annotation in order to bind to the claim.
+	// The value must be the name of the claim being bound to.
+	// This is an experimental feature and likely to change in the future.
+	createdForKey = "persistent-volume-provisioning.experimental.kubernetes.io/created-for"
```

Comment:
I can move them if that's where we'd prefer annotations to live.  Do we care about scoping at all?  These constants package protected and aren't relevant outside of this narrow functionality.

The docs explaining this feature will require the keys, of course.